plugins/metrics/pkg/agent/runner.go

```diff
@@ -195,18 +191,25 @@ func (tr *taskRunner) doPush(ctx context.Context, writeRequest *prompb.WriteRequ
 			default:
 			}
 
-			if !isRecoverable(err) {
+			switch {
+			case strings.Contains(err.Error(), "context cancelled"),
```

Comment:
For the cause behind this specific error handling, we could probably even take out this check for `context cancelled` since that specific check was kinda-sorta just a workaround for this bug.